crates/red_knot_python_semantic/src/symbol.rs

```diff
@@ -667,9 +677,33 @@ fn symbol_from_bindings_impl<'db>(
         |BindingWithConstraints {
              binding,
              narrowing_constraint,
-             visibility_constraint,
+             mut visibility_constraint,
          }| {
-            let binding = binding?;
+            let mut binding = binding?;
+
+            if binding.kind(db).is_star_import() {
+                let use_def_map = use_def_map(db, scope);
+
+                while let Some(star_import) = binding.kind(db).as_star_import() {
+                    if resolve_star_import_definition(
+                        db,
+                        scope.file(db),
+                        star_import,
+                        &symbol_table(db, scope),
+                    )
+                    .ok()
+                    .is_none_or(|symbol| symbol.symbol.is_unbound())
```

Comment:
If the star-import result is possibly-unbound, then you would need to union the type from the star import resolution with the type obtained from the prior bindings (if you take the first option below) or keep both the star-import binding and the prior bindings in this loop (if you take the flatten option below).